### ISSUE TITLE
Adguard prefers IPv4 instead of IPv6 on domain-based DNS upstreams

### ISSUE THREAD

**Issue by LucasMZReal** [AdguardTeam/AdGuardHome#5934]
### Prerequisites

- [X] I have checked the [Wiki](https://github.com/AdguardTeam/AdGuardHome/wiki) and [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) and found no answer

- [X] I have searched other issues and found no duplicates

- [X] I want to report a bug and not [ask a question or ask for help](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a)

- [X] I have set up AdGuard Home correctly and [configured clients to use it](https://github.com/AdguardTeam/AdGuardHome/wiki/Clients). (Use the [Discussions](https://github.com/AdguardTeam/AdGuardHome/discussions/categories/q-a) for help with installing and configuring clients.)

### Platform (OS and CPU architecture)

Windows/AMD64 (aka x86_64)

### Installation

GitHub releases or script from README

### Setup

On one machine

### AdGuard Home version

v0.107.32

### Action

Set up Cloudflare as a primary DNS using a domain name.

### Expected result

Expected it to work, and to use IPv6 to communicate to the server just as using a raw IP works.

### Actual result

It seems as if Adguard Home decides to prefer IPv4 when using a domain instead of an IP. Domain names are needed for tls://, https://, h3://, etc. 

The way I noticed this is because the time to get the response is consistent with my ping times depending on IPv4 and IPv6.
I have a 6ms~ connection to Cloudflare using IPv6, but a 22ms one using IPv4. These numbers are consistent with the times reported by Adguard Home, and when checking, when I'm using a raw IPv6, it works and I get about that response time for DNS requests, and when checking the higher timings, it's always when using a domain name.

I hoped to use Adguard Home as somewhat a DNS proxy from v6 to v4 because of this difference in response times between the two protocols, but I currently cannot do it while using a secure connection to my DNS provider.

### Additional information and/or screenshots

![image](https://github.com/AdguardTeam/AdGuardHome/assets/55422065/cfc5f4f6-436d-40c4-a3fd-b7eaaffddd68)
![image](https://github.com/AdguardTeam/AdGuardHome/assets/55422065/f43228be-3ad6-4f9e-b439-40ee185aa7c7)
![image](https://github.com/AdguardTeam/AdGuardHome/assets/55422065/ade966b2-6ede-485c-9b25-688b5cb3e559)

**Comment by gkhull** [comment]
It prefers whichever one is faster. "Query one upstream server at a time. AdGuard Home uses its **weighted random algorithm to pick the server so that the fastest server is used more often**." Please read before submitting issues.

**Comment by ainar-g** [comment]
Have you tried using the `bootstrap_prefer_ipv6` option described in the [Configuration file documentation](https://github.com/AdguardTeam/AdGuardHome/wiki/Configuration#configuration-file)?

**Comment by LucasMZReal** [comment]
Oh! That seems to have fixed it! Thank you! (I can see now there's another issue talking about this same thing just in other words, oh well at least now it's easier to find)

Though I would like to know why by default IPv4 is preferred? That's a weird default to have in a world where IPv6 is preferred everywhere else. A prefer IPv4 option to me is needed for the setups where the ISP has bad IPv6 peering; but it being the default is a bit off to me. People usually have the same or better peering with IPv6.

Adding this info to the FAQ or highlight it would be nice, a lot of people would prefer their traffic going through IPv6 but won't notice it's going over V4 because their peering is similar, but they might be on CGNAT which is a point of failure.

**Comment by ainar-g** [comment]
We'll probably add it to the Web UI at some point.  If the setting works for you, I'll close the issue, if you don't mind.
